profiler/profiler.api/src/org/netbeans/modules/profiler/api/ProjectUtilities.java

```diff
@@ -137,17 +135,9 @@ public static void removeOpenProjectsListener(ChangeListener listener) {
     public static Provider[] getSortedProjects(Provider[] projects) {
         List<Provider> projectsArray = Arrays.asList(projects);
 
-        Collections.sort(projectsArray,
-                new Comparator() {
-
-                    @Override
-                    public int compare(Object o1, Object o2) {
-                        Provider p1 = (Provider) o1;
-                        Provider p2 = (Provider) o2;
-
-                        return getDisplayName(p1).toLowerCase().compareTo(getDisplayName(p2).toLowerCase());
-                    }
-                });
+        projectsArray.sort((Object o1, Object o2) -> 
+            getDisplayName((Provider) o1).toLowerCase().compareTo(getDisplayName((Provider) o2).toLowerCase())
```

Comment:
Can't use `compareToIgnoreCase`?

Reply:
oh interesting, I didn't know that this method existed. I did only transform to `equalsIgnoreCase()`

I might do this at a later point since this PR is already large enough.
But i am going to add this to my hints collection for sure https://github.com/mbien/jackpot-inspections
thanks!

Reply:
looking closer at the method it likely has unintended side effects since it sorts the array in-place, and then it returns a copy via `toArray()` (which makes no sense).

other implementations do copy first, sort it and return the copy:
https://github.com/search?q=repo%3Aapache%2Fnetbeans%20getSortedProjects&type=code

should I change it to:
```java
    public static Provider[] getSortedProjects(Provider[] projects) {
        Provider[] sorted = Arrays.copyOf(projects, projects.length);
        Arrays.sort(sorted, (p1, p2) -> getDisplayName(p1).compareToIgnoreCase(getDisplayName(p2)));
        return sorted;
    }
```
@matthiasblaesing ?

Reply:
added it as extra commit